include/sparrow/buffer_view.hpp

```diff
@@ -47,14 +47,14 @@ namespace sparrow
 
         explicit buffer_view(buffer<T>& buffer);
         buffer_view(pointer p, size_type n);
-        // TODO: To be complete we also need a subrange(...) function
-        // and a constructor that takes a pair of iterators (they should
-        // be template parameters constrained by concept, not buffer_view::iterator).
-        // These are often used with view types (see std::span, which also provides
-        // first and last subview functions).
 
-        bool empty() const noexcept;
-        size_type size() const noexcept;
+        template <class It>
+            requires std::input_iterator<It>
+                     && std::same_as<std::remove_const_t<std::iter_value_t<It>>, std::remove_const_t<T>>
+        buffer_view(It first, It last);
```

Comment:
I think it's preferable that `last` is allowed to be a different type than `It` because it could be a sentinel. Then to simplify this and the other functions with these requirements, probably a concept  `input_iterator_of<T>` or something like that could help?

Same remarks for the range constructor.

Reply:
As noted below, if `buffer_view` assumes it's necessarilly a `buffer`'s content being looked at then I suspect it should be a `contiguous_iterator` instead of `input_iterator` and that would fix my previous remark.

Reply:
You' right, `contiguous_iterator` is better in this case, I also added a requirement for the sentinel